pkg/querier/http_test.go

```diff
@@ -342,3 +468,119 @@ func TestSeriesVolumeHandler(t *testing.T) {
 		require.Equal(t, time.Second.Milliseconds(), request.Step)
 	})
 }
+
+func TestResponseFormat(t *testing.T) {
+	for _, tc := range []struct {
+		url             string
+		accept          string
+		handler         func(api *QuerierAPI) http.HandlerFunc
+		result          logqlmodel.Result
+		expectedRespone string
+	}{
+		{
+			url: "/api/prom/query",
+			handler: func(api *QuerierAPI) http.HandlerFunc {
+				return api.LogQueryHandler
+			},
+			result: logqlmodel.Result{
+				Data: logqlmodel.Streams{
+					logproto.Stream{
+						Entries: []logproto.Entry{
+							{
+								Timestamp: time.Unix(0, 123456789012345),
+								Line:      "super line",
+							},
+						},
+						Labels: `{foo="bar"}`,
+					},
+				},
+				Statistics: statsResult,
+			},
+			expectedRespone: `{
+				` + statsResultString + `,
+				"streams": [
+				  {
+				    "labels": "{foo=\"bar\"}",
+				    "entries": [
+				      {
+				        "line": "super line",
+				        "ts": "1970-01-02T10:17:36.789012345Z"
+				      }
+				    ]
+				  }
+				]
+			}`,
+		},
+		{
+			url: "/loki/api/v1/query_range",
+			handler: func(api *QuerierAPI) http.HandlerFunc {
+				return api.RangeQueryHandler
+			},
+			result: logqlmodel.Result{
+				Data: logqlmodel.Streams{
+					logproto.Stream{
+						Entries: []logproto.Entry{
+							{
+								Timestamp: time.Unix(0, 123456789012345),
+								Line:      "super line",
+							},
+						},
+						Labels: `{foo="bar"}`,
+					},
+				},
+				Statistics: statsResult,
+			},
+			expectedRespone: `{
```

Comment:
Are we only checking that JSON works?

Reply:
Yes. Protobuf is already tested in the parity check. Here we test that the wiring works for the legacy format. Come to think we should probably remove the old endpoints with the next mayor release.